[RNMobile] FlatList inside the bottom-sheet

dratwas opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
In this PR i added a possibility to use FlatList inside Bottom-sheet w/o losing the swipe to close functionality (iOS) and w/o losing the optimization of FLatList.

Because we used the ScrollView to wrap content inside the BottomSheet we closed the optimization of FlatList. In this PR I added a prop called `isChildrenScrollable` and add ScrollView wrapper depends on it. I pass also all props that need to be passed to the nested FlatList by BottomSheetConsumer.

I demonstrated it on the inserter menu since it was a FlatList with `scrollEnabled` set to false.

## How has this been tested?
- Run app in develop mode
- Open inserter
- There should not be the warning `WARN  VirtualizedLists should never be nested inside plain ScrollViews with the same orientation - use another VirtualizedList-backed container instead.
` in the console

## Types of changes
Add possibility to use FlatList inside bottom-sheet

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

dratwas committed:
use optimised FlatList inside bottom-sheet

mkevins reviewed (commented):
I tested this via the steps described on Android (Pixel 3a) and iOS (via simulator iPhone 11 Pro), and the warning is no longer logged. I tested out the bottomsheet via the inserter, as well as some of the more complex flows, such as the color settings in button, and mostly everything works as expected.

Also, I've tested this via inclusion of the changeset in a draft PR here: https://github.com/WordPress/gutenberg/pull/23922, and it seems to be working there as well.

However, I noticed one odd regression in the scroll behavior of the inserter: when the drag start happens between the buttons, the gesture is lost:

<details>
<summary>Swipe gesture is lost between buttons</summary>

<img src="https://user-images.githubusercontent.com/8507675/87750416-bba67f80-c83e-11ea-97ba-4b14acba739c.gif" width="360">
</details>

I also encountered a subtle unrelated bug while testing (also observed on master), noting here since there is related work underway with navigation:

In the colors subsheet, after using the "soft" back button (the one in the subsheet header), two button presses are required to dismiss the main modal via the hardware back button:

<details>
<summary>Back button requires two presses after using "soft" back button in subsheet</summary>

<img src="https://user-images.githubusercontent.com/8507675/87750629-50a97880-c83f-11ea-9dd1-5210be9ce898.gif" width="360">
</details>

dratwas committed:
fix android scroll on separator item

dratwas commented:
Hey @mkevins thanks for your review :)
> However, I noticed one odd regression in the scroll behavior of the inserter: when the drag start happens between the buttons, the gesture is lost:

I already fixed it, could you please check it one more time?

>In the colors subsheet, after using the "soft" back button (the one in the subsheet header), two button presses are required to dismiss the main modal via the hardware back button:

Thanks for reporting, i already reimplemented the back button functionality and it seems to work on the navigation PR

mkevins reviewed (approved):
Tested again after the latest fix, and the gesture is no longer lost between buttons :tada: . LGTM!

lukewalczak commented:
I've noticed one regression in an inserter items layout, which should be left-aligned. To illustrate I attached screenshots:

before | after changes
--- | ---
<img width="493" alt="Screenshot 2020-07-23 at 09 18 19" src="https://user-images.githubusercontent.com/22746080/88261052-ddcc5180-ccc5-11ea-975c-7c9075c976d4.png"> | <img width="493" alt="Screenshot 2020-07-23 at 09 18 02" src="https://user-images.githubusercontent.com/22746080/88261054-df961500-ccc5-11ea-9f20-14facbb93cd9.png">

dratwas committed:
fix bottom spacing in inserter and justifyContent

dratwas committed:
add crollableContentStyle and use it

dratwas committed:
remove unused style

dratwas committed:
render spacing view only if needed

dratwas commented:
Thanks @lukewalczak for your report! I already fixed that issue. I also added the correct padding bottom to the inserter menu. Could you please check it one more time? :)

lukewalczak commented on packages/components/src/mobile/bottom-sheet/index.native.js:
```diff
@@ -246,6 +246,15 @@ class BottomSheet extends Component {
+		const { safeAreaBottomInset } = this.state;
+		return {
+			flexGrow: 1,
+			paddingBottom:
```
```suggestion
			paddingBottom:
				( safeAreaBottomInset || 0 ) +
				styles.scrollableContent.paddingBottom,
```

dratwas committed:
Update packages/components/src/mobile/bottom-sheet/index.native.js

Co-authored-by: Luke Walczak <lukasz.walczak.pwr@gmail.com>

dratwas committed:
fix syntax error after applying suggestion
